frappe/event_streaming/doctype/event_producer/event_producer.py

```diff
@@ -55,7 +55,8 @@ def on_update(self):
 			self.reload()
 
 	def check_url(self):
-		frappe.utils.validate_url(self.producer_url, throw=True)
+		valid_url_schemes = ("http", "https", "ftp", "ftps")
```

Comment:
You should set these as default schemes in the `validate_url` function directly, if nothing is passed. Instead of having these passed for every call.

Reply:
That again makes the function specific. Suppose, what if I want to use this function to check only for https? 

The specific schemes are better with the logic (oauth etc.) IMO

Reply:
Tomorrow if I write a new module, that requires just FTP and NTP schemes, this function won't work then.

Reply:
I was thinking more along the lines of:

```python
def validate_url(txt, throw=False, valid_schemes=None):
	if not valid_schemes:
		valid_schemes = ("http", "https", "ftp", "ftps")

	# process url
```

You already optionally accept the schemes, it's just an addition to set the default schemes.

With that change, you can just do:

```python
validate_url(url_string, valid_schemes=("https"))
```

to only check for `https`.

Reply:
Think about when I want to validate scheme agnostic, will the above work?

Reply:
I guess that's fair.